engine/api/src/main/scala/pl/touk/nussknacker/engine/api/expression/Expression.scala

```diff
@@ -35,6 +35,8 @@ case class ValueWithLazyContext[T](value: T, lazyContext: LazyContext)
 
 sealed trait TypedValue
 
-case class TypedExpression(expression: Expression, returnType: TypingResult) extends TypedValue
+case class TypedExpression(expression: Expression, returnType: TypingResult, typingInfo: ExpressionTypingInfo) extends TypedValue
 
 case class TypedExpressionMap(valueByKey: Map[String, TypedExpression]) extends TypedValue
+
+trait ExpressionTypingInfo
```

Comment:
maybe some comment how it can be used?

Reply:
do we want to assume it can be displayed as json?

Reply:
Some comment will be good. Regards to proposed assumption - I don't think so. I want to use it on backend side. It could be usefull if we could compute partial Typing for not completed expressions. Then we could move completion logic from FE to BE and do much more stuff. But for now it will be only helpful if whole expression is correct.